src/lib/recurringPurposes.js

```diff
@@ -0,0 +1,175 @@
+import log from 'cozy-logger'
+
+import { getManualPurpose, getAutomaticPurpose } from 'src/lib/trips'
+import { setAutomaticPurpose } from 'src/lib/timeseries'
+import {
+  builTimeserieQueryByAccountIdAndStartPlaceAndEndPlace,
+  queryTimeserieByDocId
+} from 'src/queries/queries'
+
+// Similar trips = same start/end dipslay name
+const findSimilarTimeseries = async (client, timeserie) => {
+  const accountId = timeserie.cozyMetadata.sourceAccount
+  const startPlaceDisplayName = timeserie.aggregation.startPlaceDisplayName
+  const endPlaceDisplayName = timeserie.aggregation.endPlaceDisplayName
```

Comment:
Are we sure that we ALWAYS have these informations? Should we be more defensive?

Reply:
We are sure about the `timeserie.aggregation` because we check it before this call, it's true we could/should be more defensive overall

Reply:
[0ec3ac0](https://github.com/cozy/coachCO2/pull/174/commits/0ec3ac02cf5a777fdb351dd28a95d56a23588af1)